src/DataProtection/DataProtection/src/TypeForwardingActivator.cs

```diff
@@ -64,6 +62,27 @@ internal object CreateInstance(Type expectedBaseType, string originalTypeName, o
         }
 
         protected string RemoveVersionFromAssemblyName(string forwardedTypeName)
-            => _versionPattern.Replace(forwardedTypeName, "");
+        {
+            // Type, Assembly, Version={Version}, Culture={Culture}, PublicKeyToken={Token}
+
+            var versionStartIndex = forwardedTypeName.IndexOf("Version=", StringComparison.Ordinal);
+            while (versionStartIndex != -1)
+            {
+                var versionEndIndex = forwardedTypeName.IndexOf(',', versionStartIndex + "Version=".Length);
+
+                if (versionEndIndex == -1)
+                {
+                    // No end index?
+                    return forwardedTypeName;
+                }
+
+                forwardedTypeName = forwardedTypeName.Substring(0, versionStartIndex - 1) + forwardedTypeName.Substring(versionEndIndex + 1);
```

Comment:
If you use the indexes correctly, you should be able to avoid the substring until you return.

Reply:
I'm not sure I understand the proposed change, do you mean to try and avoid extra string allocations, since this is just doing a while loop stripping out each version string until there's no more because Type can nested types with Version strings in it.